<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>
SPDX-FileCopyrightText: none
SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
 </author>
 <class>ClipJobManager_UI</class>
 <widget class="QDialog" name="ClipJobManager_UI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>853</width>
    <height>715</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="0" column="1" rowspan="6">
    <widget class="QGroupBox" name="param_box">
     <property name="title">
      <string>Job parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Executable</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="3">
       <widget class="KUrlRequester" name="url_binary"/>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="KMessageWidget" name="script_message">
        <property name="closeButtonVisible">
         <bool>false</bool>
        </property>
        <property name="messageType">
         <enum>KMessageWidget::Warning</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QToolButton" name="info_button">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset theme="help-about">
          <normaloff>.</normaloff>.</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Arguments</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="3">
       <widget class="QPlainTextEdit" name="job_params"/>
      </item>
      <item row="5" column="0" colspan="3">
       <widget class="KMessageWidget" name="info_message">
        <property name="text">
         <string>In the arguments, the following options are available: &lt;ul&gt;&lt;li&gt;&lt;b&gt;{source}&lt;/b&gt; will be replaced by the Bin Clip source file path&lt;/li&gt;&lt;li&gt;&lt;b&gt;{output}&lt;/b&gt; will be replaced by the output file name&lt;/li&gt;&lt;li&gt;&lt;b&gt;{param1}&lt;/b&gt; will be replaced by the first parameter value&lt;/li&gt;&lt;li&gt;&lt;b&gt;{param2}&lt;/b&gt; will be replaced by the second parameter value&lt;/li&gt;&lt;/ul&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="closeButtonVisible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Parameter 1 {param1}</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="1" rowspan="5">
          <widget class="QPlainTextEdit" name="param1_list">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="placeholderText">
            <string>List of options</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QRadioButton" name="param1_islist">
           <property name="text">
            <string>Request option from list</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="0">
          <widget class="QLineEdit" name="param1_name">
           <property name="placeholderText">
            <string>Parameter description</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QRadioButton" name="param1_isfile">
           <property name="text">
            <string>Request file path</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QRadioButton" name="param1_isframe">
           <property name="text">
            <string>Current clip frame</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="6" column="2">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Parameter 2 {param2}</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="3" column="0">
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="1" rowspan="4">
          <widget class="QPlainTextEdit" name="param2_list">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="placeholderText">
            <string>List of options</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLineEdit" name="param2_name">
           <property name="placeholderText">
            <string>Parameter description</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QRadioButton" name="param2_islist">
           <property name="text">
            <string>Request option from list</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QRadioButton" name="param2_isfile">
           <property name="text">
            <string>Request file path</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="7" column="0" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Output file extension</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="destination_pattern"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QPlainTextEdit" name="taskDescription">
     <property name="placeholderText">
      <string>Optional description that will show up if a parameter is requested</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Job Description</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QToolButton" name="button_add">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset theme="list-add">
         <normaloff>../../../../../../.designer/backup</normaloff>../../../../../../.designer/backup</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="button_delete">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset theme="edit-delete">
         <normaloff>../../../../../../.designer/backup</normaloff>../../../../../../.designer/backup</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Close</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="5" column="0">
    <widget class="QGroupBox" name="folder_box">
     <property name="title">
      <string>Project Bin behavior on completion</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0" colspan="2">
       <widget class="QRadioButton" name="radio_replace">
        <property name="text">
         <string>Replace original with result</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="bin_subfolder">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Use a subfolder</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="radio_folder">
        <property name="text">
         <string>Put resulting clip in a folder</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QComboBox" name="combo_folder">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>Placed at top level</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Placed in the original clip folder</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLineEdit" name="folder_name">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="placeholderText">
         <string>Folder name</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QListWidget" name="job_list">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Custom clip jobs</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QGroupBox" name="clip_types">
     <property name="title">
      <string>Applies to</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="enable_video">
        <property name="text">
         <string>Video</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="enable_audio">
        <property name="text">
         <string>Audio</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="enable_image">
        <property name="text">
         <string>Images</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QWidget</extends>
   <header>kurlrequester.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KMessageWidget</class>
   <extends>QFrame</extends>
   <header>kmessagewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ClipJobManager_UI</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ClipJobManager_UI</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radio_folder</sender>
   <signal>toggled(bool)</signal>
   <receiver>bin_subfolder</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>390</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>407</x>
     <y>563</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radio_folder</sender>
   <signal>toggled(bool)</signal>
   <receiver>combo_folder</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bin_subfolder</sender>
   <signal>toggled(bool)</signal>
   <receiver>folder_name</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>407</x>
     <y>563</y>
    </hint>
    <hint type="destinationlabel">
     <x>407</x>
     <y>596</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>info_button</sender>
   <signal>toggled(bool)</signal>
   <receiver>info_message</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>678</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>param1_islist</sender>
   <signal>toggled(bool)</signal>
   <receiver>param1_list</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>360</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>param2_islist</sender>
   <signal>toggled(bool)</signal>
   <receiver>param2_list</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>360</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>541</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
